File: thisweek_masuipeo2/tyama_codeiq3082.rb
#!/usr/bin/env ruby
def comb(n,k)
	return 0 if k<0||n<k
	r=1
	k=n-k if k>n/2
	k.times{|i|
		r=r*(n-i)/(i+1)
	}
	r
end
N,A,B=gets.split.map(&:to_i)
p comb(N-2,A-2)*comb(A-2,B-2)

File: thisweek_masuipeo2/tyama_codeiq3082.hs
#!/usr/bin/env runghc
import Control.Applicative
import Data.List
comb n k = if k<0||n<k then 0 else foldl' (\r i -> (r*(n-i) `div` (i+1))) 1 [0..k-1]
main = (map read . words) <$> getLine >>= \[n,a,b] -> print $ (comb (n-2) (a-2)) * (comb (a-2) (b-2))
